packages/react-router-dom/modules/Link.js

```diff
@@ -52,9 +57,24 @@ class Link extends React.Component {
     }
   }
 
-  render() {
-    const { replace, to, ...props } = this.props // eslint-disable-line no-unused-vars
+  absolute() {
+    const { to } = this.props
+    const { match } = this.context.router
+    const base = (match && match.url) ? match.url : ''
+    return resolveLocation(to, base)
+  }
+
+  componentWillMount() {
+    this.unlisten = this.context.router.listen(() => { this.forceUpdate() })
```

Comment:
wait, @pshrmn, YOU'RE ON TWITTER?! I was looking for you the other day but couldn't find you!

I'm not a huge fan of HOCs, no, and my recent work on the router confirmed yet again why I don't really like them. They just make things more complicated. I was using an HOC for earlier versions of `<Route>` and `<Switch>`, but things were much simpler when I just inlined the code in `withRouter` into those components. Now we give people a `withRouter` HOC, but I honestly don't know if I'll ever use it.

As a side note: one thing this PR really, really needs is some good examples that really demonstrate this functionality and put it through its paces. Every time I've tried to work on this problem I have to build a real app with it before I can see all the things I'm doing wrong.